Add masterless state.highstate integration test

Ch3LL opened the pull request:
### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/43539

Might need to be rebased if https://github.com/saltstack/salt/pull/44613 is merged first.

Ch3LL committed:
Add masterless state.highstate integration test

rallytime commented:
@Ch3LL There is a related test failure on this. Can you swing by and fix it up?

Ch3LL committed:
Change check to the state id

rallytime commented on tests/support/case.py:
```diff
@@ -210,8 +210,10 @@ def run_cp(self, arg_str, with_retcode=False, catch_stderr=False):
 
-    def run_call(self, arg_str, with_retcode=False, catch_stderr=False):
-        arg_str = '--config-dir {0} {1}'.format(self.get_config_dir(), arg_str)
+    def run_call(self, arg_str, local=False, with_retcode=False, catch_stderr=False):
```
The `local` kwarg addition should go at the end of the list so we don't cause an API error. :)

rallytime commented on tests/support/case.py:
```diff
@@ -549,11 +551,12 @@ def run_cp(self, arg_str, with_retcode=False, catch_stderr=False):
                                timeout=60)
 
-    def run_call(self, arg_str, with_retcode=False, catch_stderr=False):
+    def run_call(self, arg_str, local=False, with_retcode=False, catch_stderr=False):
```
Same comment here.

Ch3LL commented:
I rebased this branch with 2017.7 which had the correct changes for moving `local` to the end of the kwarg list on the run_call methods. So the files is now removed and shouldn't be an issue. Here's to the hope that the tests pass :)
